patches/api/0418-More-native-map-PDC-methods-use-adventure-Key.patch

```diff
@@ -0,0 +1,290 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: RyGuy <git@ryguy.me>
+Date: Tue, 28 Mar 2023 01:27:28 -0400
+Subject: [PATCH] More native map PDC methods, use adventure Key
+
+This commit has PDC inherit Iterable<NamespacedKey>, replaces all NamespacedKey uses with adventure's key,
+and implements these methods
+- #setIfAbsent(NamespacedKey, PersistentDataType, Object)
+- #computeIfAbsent(NamespacedKey, PersistentDataType, Function)
+- #computeIfPresent(NamespacedKey, PersistentDataType, BiFunction)
+- #compute(NamespacedKey, PersistentDataType, BiFunction)
+- #getOrThrow(NamespacedKey, PersistentDataType, Supplier)
+- #getOrThrow(NamespacedKey, PersistentDataType)
+- #replace(NamespacedKey, PersistentDataType, Object, Object)
+- #replace(NamespacedKey, PersistentDataType, Object)
+
+diff --git a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+index d30ad7cae1dfbb47ab1a6e3e2fcd6fb76163fe78..78dc690d785e756ac4752e53dd95c164556d0cff 100644
+--- a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
++++ b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+@@ -1,15 +1,22 @@
+ package org.bukkit.persistence;
+ 
+-import java.util.Set;
+ import org.bukkit.NamespacedKey;
+ import org.jetbrains.annotations.NotNull;
+ import org.jetbrains.annotations.Nullable;
+ 
++import java.util.Iterator;
++import java.util.NoSuchElementException;
++import java.util.Set;
++import java.util.function.BiFunction;
++import java.util.function.Function;
++import java.util.function.Supplier;
++
+ /**
+  * This interface represents a map like object, capable of storing custom tags
+  * in it.
+  */
+-public interface PersistentDataContainer {
++//Paper - replace all NamespacedKey uses except for #getKeys with net.kyori.adventure.key.Key
++public interface PersistentDataContainer extends Iterable<net.kyori.adventure.key.Key> {
+ 
+     /**
+      * Stores a metadata value on the {@link PersistentDataHolder} instance.
+@@ -28,11 +35,11 @@ public interface PersistentDataContainer {
+      * @throws NullPointerException if the key is null
+      * @throws NullPointerException if the type is null
+      * @throws NullPointerException if the value is null. Removing a tag should
+-     * be done using {@link #remove(NamespacedKey)}
++     * be done using {@link #remove(net.kyori.adventure.key.Key)}
+      * @throws IllegalArgumentException if no suitable adapter will be found for
+      * the {@link PersistentDataType#getPrimitiveType()}
+      */
+-    <T, Z> void set(@NotNull NamespacedKey key, @NotNull PersistentDataType<T, Z> type, @NotNull Z value);
++    <T, Z> void set(@NotNull net.kyori.adventure.key.Key key, @NotNull PersistentDataType<T, Z> type, @NotNull Z value);
```

Comment:
Yeah, there was a reason I didn't include changing this. This is an ABI break, plugins compiled against previous versions of the API will not run with this change.

Reply:
![image](https://user-images.githubusercontent.com/12160169/228735705-14ab5f7d-a22f-4285-91d4-d317127a6436.png)
When it changing only what you specified

Reply:
move the method impl to the Craft impl, and then can probably just make duplicates of all the methods as private methods, that just accept strings from the NamespacedKeys that's all they are done with anyways, the impl just calls toString on the keys.